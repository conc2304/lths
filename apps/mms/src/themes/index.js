import PropTypes from 'prop-types';
import { useMemo } from 'react';

import { CssBaseline } from '@mui/material';
import { createTheme, ThemeProvider } from '@mui/material/styles';

import getPalette from './palette';
import getTypography from './typography';
import getOverrides from './overrides';


export default function DashThemeProvider({ children }) {
    const theme = getPalette();

    const themeTypography = getTypography(`'Public Sans', sans-serif`);

    const themeOptions = useMemo(
        () => ({
            breakpoints: {
                values: {
                    xs: 0,
                    sm: 768,
                    md: 1024,
                    lg: 1266,
                    xl: 1536
                }
            },
            direction: 'ltr',
            mixins: {
                // layout header settings
                toolbar: {
                    minHeight: 50,
                    paddingTop: 4,
                    paddingBottom: 4
                },
            },
            palette: theme.palette,
            typography: themeTypography
        }),
        [theme, themeTypography]
    );

    const themes = createTheme(themeOptions);
    themes.components= getOverrides(themes);

    console.log('themes.components')
    console.log(themes.components)

    return (<ThemeProvider theme={themes}>
                <CssBaseline />
                {children}
            </ThemeProvider>);
}

DashThemeProvider.propTypes = {
    children: PropTypes.node
};
