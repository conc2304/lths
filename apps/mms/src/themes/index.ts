export { DashThemeProvider } from '@lths-mui/shared/mui-themes';
