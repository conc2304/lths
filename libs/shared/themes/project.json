{
  "name": "shared-themes",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/themes/src",
  "projectType": "library",
  "tags": ["shared", "themes"],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/shared/themes/**/*.{ts,tsx,js,jsx}"]
      }
    }
  }
}
