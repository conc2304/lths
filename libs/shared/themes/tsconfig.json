{
  "compilerOptions": {
    "jsx": "react-jsx",
    "allowJs": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true
  },
  "files": [
    "src/createPalette.d.ts",
    "src/Button.d.ts"
  ],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    }
  ],
  "extends": "../../../tsconfig.base.json"
}