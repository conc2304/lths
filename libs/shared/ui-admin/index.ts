export * from './src/feature-flags';
