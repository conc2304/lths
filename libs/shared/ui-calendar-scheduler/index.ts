export * from './src/lib/components/calendar/index';
export * from './src/lib/components/mock-events';
