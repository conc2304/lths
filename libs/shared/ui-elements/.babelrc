{
  "presets": [
    "@babel/preset-env",
    "@babel/preset-typescript",
    [
      "@nx/react/babel",
      {
        "runtime": "automatic",
        "useBuiltIns": "usage"
      }
    ]
  ],
  "plugins": []
}
