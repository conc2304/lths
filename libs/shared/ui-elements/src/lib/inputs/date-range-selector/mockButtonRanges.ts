import { subDays, subHours, subMonths } from 'date-fns';

export const now = new Date('December 17, 1995 03:24:00');
export const ButtonGroupConf = [
  {
    label: '1 Hour',
    value: subHours(now, 1),
  },
  {
    label: '1 Day',
    value: subDays(now, 1),
  },
  {
    label: '7 Days',
    value: subDays(now, 7),
  },
  {
    label: '30 Days',
    value: subDays(now, 30),
  },
  {
    label: '3 Months',
    value: subMonths(now, 3),
  },
  {
    label: '6 Months',
    value: subMonths(now, 6),
  },
  {
    label: '12 Months',
    value: subMonths(now, 12),
  },
];