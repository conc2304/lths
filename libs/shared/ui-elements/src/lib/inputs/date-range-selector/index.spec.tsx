import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { RBThemeProvider } from '@lths-mui/shared/themes';
import { DateRangeSelector } from './index';
import { ButtonGroupConf } from './mockButtonRanges';

describe('Date Range Selector', () => {
  let themedComponent: RenderResult;
  let component: JSX.Element;

  beforeEach(() => {
    const component = (
      <DateRangeSelector
        dateOptions={ButtonGroupConf}
        onChange={({ startDate, endDate }) =>
          console.log('changed', { startDate, endDate })
        }
      />
    );
    themedComponent = render(RBThemeProvider({ children: component }));
  });
  it('should render', () => {
    const { baseElement } = themedComponent;
    expect(baseElement).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { baseElement } = themedComponent;
    expect(baseElement).toMatchSnapshot();
  });
});
