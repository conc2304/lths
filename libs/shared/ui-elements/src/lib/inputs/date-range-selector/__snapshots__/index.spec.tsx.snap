// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeSelector should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="LthsDateRangeSelector-root MuiBox-root css-0"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-nr2g4u-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-md-auto MuiGrid2-grid-xs-12 css-1jlbob8-MuiGrid2-root"
        >
          <div
            aria-label="Predifinded Date Range Fiter"
            class="MuiToggleButtonGroup-root Lths-Button-Group css-1afgqwa-MuiToggleButtonGroup-root"
            role="group"
          >
            <button
              aria-label="1 Hour"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="1-hour"
            >
              1 Hour
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="1 Day"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="1-day"
            >
              1 Day
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="7 Days"
              aria-pressed="true"
              class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="7-days"
            >
              7 Days
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="30 Days"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="30-days"
            >
              30 Days
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="3 Months"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="3-months"
            >
              3 Months
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="6 Months"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="6-months"
            >
              6 Months
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="12 Months"
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-info MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-irth4x-MuiButtonBase-root-MuiToggleButton-root"
              role="button"
              tabindex="0"
              type="button"
              value="12-months"
            >
              12 Months
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle MuiDivider-vertical css-15n1gql-MuiDivider-root"
        />
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-md-auto MuiGrid2-grid-xs-12 css-1ktj8a9-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1pxy8fg-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-md-6 MuiGrid2-grid-xs-6 css-wsndb2-MuiGrid2-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root Lths-Date-Picker css-d4fgww-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-rrfoq5-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r4:"
                  id=":r4:-label"
                >
                  START
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1q4o57m-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r4:"
                    inputmode="text"
                    placeholder="⁦⁨MM⁩ / ⁨DD⁩ / ⁨YYYY⁩⁩"
                    type="text"
                    value="⁦⁨05⁩ / ⁨07⁩ / ⁨2023⁩⁩"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date, selected date is May 7, 2023"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarMonthOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zm0-12H5V6h14v2zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        START
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-md-4 MuiGrid2-grid-xs-4 css-1uoi70d-MuiGrid2-root"
            >
              <div
                class="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root Lths-Date-Picker css-d4fgww-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-rrfoq5-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for=":r6:"
                  id=":r6:-label"
                >
                  END
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1q4o57m-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r6:"
                    inputmode="text"
                    placeholder="⁦⁨MM⁩ / ⁨DD⁩ / ⁨YYYY⁩⁩"
                    type="text"
                    value="⁦⁨05⁩ / ⁨13⁩ / ⁨2023⁩⁩"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="Choose date, selected date is May 13, 2023"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CalendarMonthOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zm0-12H5V6h14v2zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        END
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row css-27y2fm-MuiGrid2-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondaryButton MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondaryButton MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-h4u5m5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            UPDATE PERIOD
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
